packages/instantsearch.js/src/middlewares/__tests__/createRouterMiddleware.ts

```diff
@@ -102,4 +104,41 @@ describe('router', () => {
       }
     `);
   });
+
+  it('does not update ui state with onUpdate if no widgets have been added', async () => {
+    const state = {
+      'my-index': {
+        query: 'iPhone',
+      },
+    };
+
+    const searchClient = createSearchClient();
+    const router: Router = {
+      onUpdate: jest.fn((callback) => {
+        callback(state);
+      }),
+      read: () => state,
+      write: () => {},
+      createURL: () => '',
+      dispose: () => {},
+    };
+
+    const search = instantsearch({
+      indexName: 'my-index',
+      searchClient,
+      routing: {
+        router,
+      },
+    });
+
+    const setUiStateSpy = jest.spyOn(search, 'setUiState');
+
+    search.mainIndex.getHelper = jest.fn(() => search.mainHelper);
```

Comment:
I couldn't figure out why the main index widget would not return a correct helper through this method in the test suite, so this makes it work, but it's not ideal.

Reply:
When this line is commented out, the test still passes for me though

Reply:
You're right! I cleaned my repo and tested again, it works. I don't know what might have had caused this earlier. I'll remove the line thanks!